ANPR/src/anpr/license_plate_recognition.py

```diff
@@ -0,0 +1,283 @@
+# import the necessary packages
+import re
+
+import numpy as np
+
+from ocr import *
+
+class LicensePlateResult:
+    """
+    This class contains the result of a license plate recognition detection algorithm. It stores the detected text and
+    the location on the image. It also provides a function to find a given license plate format in the detected text.
+    """
+
+    def __init__(self, text: str, location: ResultLocation):
+        self.text = text
+        self.location = location
+
+    def match_format(self, formats: list):
```

Comment:
You can define the output type of the function like so: `def match_format(self, formats: list) -> list[LicensePlateResult`.